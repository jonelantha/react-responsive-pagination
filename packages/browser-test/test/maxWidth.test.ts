import { setupThrowOnError } from './helper';

beforeAll(async () => {
  setupThrowOnError(page);

  await page.goto(harnessUrl);

  await page.setViewportSize({ width: 600, height: 700 });

  await page.fill('#totalAsJson', '100');
});

describe('Pagination maxWidth', () => {
  test.each([[null], [400], [250], [null]])(
    'renders correctly with maxWidth of %p',
    async width => {
      await page.fill('#maxWidthAsJson', (width && width.toString()) || '');

      await page.evaluate(() => new Promise(requestAnimationFrame));

      const paginationHtml = await page.$eval('ul.pagination', ul => ul.innerHTML);

      expect(paginationHtml).toMatchSnapshot();
    },
  );
});
