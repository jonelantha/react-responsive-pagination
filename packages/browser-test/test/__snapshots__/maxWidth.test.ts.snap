// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination maxWidth renders correctly with maxWidth of 250 1`] = `"<li class=\\"page-item disabled\\"><span class=\\"page-link\\"><span aria-hidden=\\"true\\">«</span><span class=\\"sr-only\\">Previous</span></span></li><li class=\\"page-item active\\"><a class=\\"page-link\\" href=\\"#\\" aria-label=\\"(current)\\"><span aria-hidden=\\"true\\">1</span><span class=\\"sr-only\\">(current)</span></a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">2</a></li><li class=\\"page-item disabled\\" aria-hidden=\\"true\\"><span class=\\"page-link\\">…</span></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">100</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\" aria-label=\\"Next\\"><span aria-hidden=\\"true\\">»</span><span class=\\"sr-only\\">Next</span></a></li>"`;

exports[`Pagination maxWidth renders correctly with maxWidth of 400 1`] = `"<li class=\\"page-item disabled\\"><span class=\\"page-link\\"><span aria-hidden=\\"true\\">«</span><span class=\\"sr-only\\">Previous</span></span></li><li class=\\"page-item active\\"><a class=\\"page-link\\" href=\\"#\\" aria-label=\\"(current)\\"><span aria-hidden=\\"true\\">1</span><span class=\\"sr-only\\">(current)</span></a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">2</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">3</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">4</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">5</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">6</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">7</a></li><li class=\\"page-item disabled\\" aria-hidden=\\"true\\"><span class=\\"page-link\\">…</span></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">100</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\" aria-label=\\"Next\\"><span aria-hidden=\\"true\\">»</span><span class=\\"sr-only\\">Next</span></a></li>"`;

exports[`Pagination maxWidth renders correctly with maxWidth of null 1`] = `"<li class=\\"page-item disabled\\"><span class=\\"page-link\\"><span aria-hidden=\\"true\\">«</span><span class=\\"sr-only\\">Previous</span></span></li><li class=\\"page-item active\\"><a class=\\"page-link\\" href=\\"#\\" aria-label=\\"(current)\\"><span aria-hidden=\\"true\\">1</span><span class=\\"sr-only\\">(current)</span></a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">2</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">3</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">4</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">5</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">6</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">7</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">8</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">9</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">10</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">11</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">12</a></li><li class=\\"page-item disabled\\" aria-hidden=\\"true\\"><span class=\\"page-link\\">…</span></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">100</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\" aria-label=\\"Next\\"><span aria-hidden=\\"true\\">»</span><span class=\\"sr-only\\">Next</span></a></li>"`;

exports[`Pagination maxWidth renders correctly with maxWidth of null 2`] = `"<li class=\\"page-item disabled\\"><span class=\\"page-link\\"><span aria-hidden=\\"true\\">«</span><span class=\\"sr-only\\">Previous</span></span></li><li class=\\"page-item active\\"><a class=\\"page-link\\" href=\\"#\\" aria-label=\\"(current)\\"><span aria-hidden=\\"true\\">1</span><span class=\\"sr-only\\">(current)</span></a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">2</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">3</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">4</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">5</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">6</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">7</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">8</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">9</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">10</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">11</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">12</a></li><li class=\\"page-item disabled\\" aria-hidden=\\"true\\"><span class=\\"page-link\\">…</span></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\">100</a></li><li class=\\"page-item\\"><a class=\\"page-link\\" href=\\"#\\" aria-label=\\"Next\\"><span aria-hidden=\\"true\\">»</span><span class=\\"sr-only\\">Next</span></a></li>"`;
